src/autopas/particles/Particle.h

```diff
@@ -7,222 +7,18 @@
-   * @param v new velocity
-   */
-  void setV(const std::array<floatType, 3> &v) { _v = v; }
-
-  /**
-   * Add a vector to the current velocity of the particle
-   * @param v vector to be added
-   */
-  void addV(const std::array<floatType, 3> &v) { _v = ArrayMath::add(_v, v); }
-
-  /**
-   * Creates a string containing all data of the particle.
-   * @return String representation.
-   */
-  virtual std::string toString() {
-    std::ostringstream text;
-    // clang-format off
-    text << "Particle"
-         << "\nID      : " << _id
-         << "\nPosition: "
-         << _r[0] << " | " << _r[1] << " | " << _r[2]
-         << "\nVelocity: "
-         << _v[0] << " | " << _v[1] << " | " << _v[2]
-         << "\nForce   : "
-         << _f[0] << " | " << _f[1] << " | " << _f[2];
-    // clang-format on
-    return text.str();
-  }
-
-  /**
-   * Defines whether the particle is owned by the current AutoPas object (aka (MPI-)process)
-   * @return true if the particle is owned by the current AutoPas object, false otherwise
-   */
-  bool isOwned() const { return _isOwned; }
-
-  /**
-   * Set the owned state to the given value
-   * @param owned
-   */
-  void setOwned(bool owned) { _isOwned = owned; }
-
-  /**
-   * Enums used as ids for accessing and creating a dynamically sized SoA.
-   */
-  enum AttributeNames : int { id, posX, posY, posZ, forceX, forceY, forceZ, owned };
-
-  /**
-   * Floating Point Type used for this particle
-   */
-  typedef floatType ParticleFloatingPointType;
-
-  /**
-   * The type for the soa storage.
-   * owned is currently used as a floatType to ease calculations within the functors.
-   */
-  typedef typename autopas::utils::SoAType<size_t, floatType, floatType, floatType, floatType, floatType, floatType,
-                                           floatType>::Type SoAArraysType;
-
-#if defined(AUTOPAS_CUDA)
-  /**
-   * The type for storage arrays for Cuda
-   */
-  typedef typename autopas::utils::CudaSoAType<size_t, floatType, floatType, floatType, floatType, floatType,
-                                               floatType>::Type CudaDeviceArraysType;
-#else
-  /**
-   * The type for storage arrays for Cuda
-   * empty if compiled without Cuda Support
-   */
-  typedef typename autopas::utils::CudaSoAType<>::Type CudaDeviceArraysType;
-#endif
-
- protected:
-  /**
-   * Particle position as 3D coordinates.
-   */
-  std::array<floatType, 3> _r;
-  /**
-   * Particle velocity as 3D vector.
-   */
-  std::array<floatType, 3> _v;
-  /**
-   * Force the particle experiences as 3D vector.
-   */
-  std::array<floatType, 3> _f;
-  /**
-   * Particle id.
-   */
-  unsigned long _id;
-
-  /**
-   * Defines whether the particle is owned by the current AutoPas object (aka (MPI-)process)
-   */
-  bool _isOwned;
-};
 /**
  * Particle with all variables in 32 bit precision
  */
-typedef ParticleBase<float> ParticleFP32;
+typedef ParticleBase<float, unsigned int> ParticleFP32;
```

Comment:
is there any reason to make this an `unsigned int`? it might introduce errors if too many particles are used (more than 4 billion).
Maybe introduce a `ParticleFP32UI` or sth with a better name?

Reply:
The idea was to make the id as wide as the floating point numbers, but I'll change that. Actually, the reason to set the id type via a template parameter was, that in huge simulations the difference between 64bit and 32bit has a significant impact on memory consumption. However, since the floating point precision doesn't depend on the size of the simulation, it might be better to follow your suggestion.